backend-project/small_eod/letters/views.py

```diff
@@ -12,3 +25,31 @@ class LetterViewSet(viewsets.ModelViewSet):
 class DescriptionViewSet(viewsets.ModelViewSet):
     queryset = Description.objects.all()
     serializer_class = DescriptionSerializer
+
+
+class FileViewSet(NestedViewSetMixin,
+                  mixins.CreateModelMixin,
+                  mixins.RetrieveModelMixin,
+                  viewsets.GenericViewSet):
+    model = File
+    serializer_class = FileSerializer
+    parent_lookup_kwargs = {
+        'letter_pk': 'letter__pk'
+    }
+
+    def perform_create(self, serializer):
+        serializer.save(letter=get_object_or_404(Letter, pk=self.kwargs['letter_pk']))
+
+class PresignedUploadFileView(APIView):
+    """
+    Generates pre-signed form data for uploading files to object storage.
+    """
+
+    serializer_class = SignRequestSerializer
+
+    @swagger_auto_schema(request_body=SignRequestSerializer)
+    def post(self, request, format=None):
```

Comment:
Trochę krócej to napisałem i w dodatku z pełną dokumentacją parametrów wejścia i wyjścia.